[Chan Zhuo Yang] iP

zhuoyang125 opened the pull request:
# Duke Task App

> "After using Duke Task, I was able to
> accomplish so much more!" - Satisfied Customer

Duke Task, a **simple** and **powerful** way to track your tasks :100:

- Ask tasks
- Delete tasks
- List tasks

Over ~~5~~ 7 people have used and LOVED our app!

Here's a sneak peek of one of our methods! This method is called `markAsCompleted()`
```java
public void markAsCompleted() {
    this.isCompleted = true;
}
```
## How to run our app?
1. Click on [this link](https://github.com/zhuoyang125/ip)
2. Run the .jar file
3. Star the fork

## Future Plans
- [ ] IPO
- [ ] Launch on Google Play

Commit:
Add Gradle support

zhuoyang125 committed:
add Level-1

zhuoyang125 committed:
Add Level-2

zhuoyang125 committed:
Add Task class

zhuoyang125 committed:
Add Level-3

zhuoyang125 committed:
add Level 4

zhuoyang125 committed:
fix edge case when duration contains /at or /by

zhuoyang125 committed:
add A-TextUiTesting

zhuoyang125 committed:
Edit output to remove redundant spaces

zhuoyang125 committed:
add A-TextUiTesting

zhuoyang125 committed:
Add A-TextUiTesting with more test cases

zhuoyang125 committed:
Merge branch 'master' of https://github.com/zhuoyang125/ip

zhuoyang125 committed:
add Level-5

zhuoyang125 committed:
add Level-5

zhuoyang125 committed:
add Level-6

zhuoyang125 committed:
add Level-6

zhuoyang125 committed:
update A-TextUiTesting test cases and regression testing

zhuoyang125 committed:
update A-TextUiTesting test cases and regression testing

zhuoyang125 committed:
Merge branch 'master' of https://github.com/zhuoyang125/ip

zhuoyang125 committed:
Add Level 7

zhuoyang125 committed:
Add Level-8

zhuoyang125 committed:
Merge branch 'branch-Level-7'

zhuoyang125 committed:
Merge branch 'branch-Level-8'

zhuoyang125 committed:
Add A-MoreOOP

zhuoyang125 committed:
Add A-Packages

zhuoyang125 committed:
Add A-JUnit

zhuoyang125 committed:
Add A-JavaDoc

zhuoyang125 committed:
Add A-CodingStandard

zhuoyang125 committed:
Add Level-9

zhuoyang125 committed:
Merge branch 'branch-A-JavaDoc'

zhuoyang125 committed:
Fix merge conflicts due to coding standard

zhuoyang125 committed:
Merge branch 'branch-Level-9'

zhuoyang125 committed:
Remove merge conflict messages

zhuoyang125 committed:
Remove conflict messages in Task

zhuoyang125 committed:
Make Task class abstract

zhuoyang125 committed:
Edit ParserTest to match Level-9 changes

atyhamos commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class Duke {
```
Shouldn't your classes have Javadoc comments?

atyhamos commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,32 @@
+    private LocalDate timePeriod;
+
+    public Event(String taskname, String timePeriod) {
+        super(taskname);
```
Shouldn't taskname be taskName in camel case?

atyhamos commented on src/main/java/duke/utils/TaskList.java:
```diff
@@ -0,0 +1,101 @@
+        return this.taskList.size();
+    }
+
+    public void search(String query) {
```
Shouldn't there be a Javadoc comment here?

atyhamos commented on src/main/java/duke/utils/Ui.java:
```diff
@@ -0,0 +1,54 @@
+ * The Ui class is responsible for methods that interact with the user.
+ */
+public class Ui {
+    private static final String GREETING = "Hi, I'm Duke, your personal assistant!\n";
```
I like that these are declared as constants, but shouldn't there be comments for them too?

atyhamos commented on src/test/java/duke/utils/TaskListTest.java:
```diff
@@ -0,0 +1,47 @@
+            assertEquals(expected, e.getMessage());
+        }
+    }
+
+
```
Should there be extra lines here?

atyhamos commented on src/main/java/duke/exceptions/InvalidCommandException.java:
```diff
@@ -0,0 +1,7 @@
+
+public class InvalidCommandException extends DukeException {
+    public InvalidCommandException() {
+        super("I'm not sure what you mean. (T_T)");
```
I like this! Very emotive 😃

atyhamos commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+
+            }
+
+
```
Should there be these extra spacings between lines?

atyhamos commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,47 @@
+ * The Task class encapsulates a task.
+ */
+public abstract class Task {
+    /** Name of the task */
```
Should these have a period at the end of the comment?

atyhamos reviewed (commented):
I like the cleanliness of your code for the most part.
Well thought-out variable and method names like loadData, isCompleted.
There are some portions where your code deviated from the coding conventions however: 
- Javadoc comments are missing for some classes and methods
- Unnecessary gaps between lines
- Did not use camel case for variable names

atyhamos commented on src/main/java/duke/utils/Parser.java:
```diff
@@ -0,0 +1,81 @@
+            } else {
+                throw new InvalidTaskIdException();
+            }
+        } else if (commandInput.matches("todo(.*?)")) {
+            if (commandInput.split(" ").length < 2) {
+                throw new EmptyTaskDescriptionException();
+            } else {
+                String taskname = commandInput.split(" ", 2)[1];
+                ToDo todo = new ToDo(taskname);
+                taskList.add(todo);
+                ui.printTasks(taskList);
+            }
+        } else if (commandInput.matches("deadline(.*?)")) {
+            if (commandInput.split(" ").length < 2) {
+                throw new EmptyTaskDescriptionException();
+            } else {
+                String firstCommand = commandInput.split("/by", 2)[0];
+                String taskname = firstCommand.split(" ", 2)[1];
+                String dueDate = commandInput.split("/by", 2)[1];
+                Deadline deadline = new Deadline(taskname.trim(), dueDate.trim());
+                taskList.add(deadline);
+                ui.printTasks(taskList);
+            }
+        } else if (commandInput.matches("event(.*?)")) {
+            if (commandInput.split(" ").length < 2) {
+                throw new EmptyTaskDescriptionException();
+            } else {
+                String firstCommand = commandInput.split("/at", 2)[0];
+                String taskname = firstCommand.split(" ", 2)[1];
+                String duration = commandInput.split("/at", 2)[1];
+                Event event = new Event(taskname.trim(), duration.trim());
+                taskList.add(event);
+                ui.printTasks(taskList);
+            }
+        } else if (commandInput.matches("find\\s(.*?)")) {
+            taskList.search(commandInput.split(" ")[1]);
+        } else {
+            throw new InvalidCommandException();
+        }
+    }
```
I'm not sure I like this big chunk of code... 🤔 
Comments would be nice as it may get quite confusing as to what's happening.

zhuoyang125 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

zhuoyang125 committed:
Add A-Gradle

zhuoyang125 committed:
Merge branch 'branch-A-Gradle'

zhuoyang125 committed:
Add Level-10

zhuoyang125 committed:
Merge branch 'branch-Level-10'

zhuoyang125 committed:
Remove redundant package imports

zhuoyang125 committed:
Add assertions

zhuoyang125 committed:
Improve code quality

zhuoyang125 committed:
Merge pull request #1 from zhuoyang125/branch-A-Assertions

Add assertions to verify assumptions in the program

zhuoyang125 committed:
Merge branch 'master' into branch-A-CodeQuality

zhuoyang125 committed:
Merge pull request #2 from zhuoyang125/branch-A-CodeQuality

Improve code quality

zhuoyang125 committed:
Set up continuous integration

zhuoyang125 committed:
Add C-DetectDuplicates

zhuoyang125 committed:
Revert "Add C-DetectDuplicates"

This reverts commit bef08e344aeab96120544a7d68be361a01209a2c.

zhuoyang125 committed:
Add C-DetectDuplicates

When adding a task to the task list, the task list will check whether
the task currently exists. If so, it will prompt the user and the task
cannot be added.

To determine if the task exists, Task objects are compared with the
equals() method. Equality of Task objects is defined as:
*ToDo: String objects for the ToDo name are equal
*Event: String and LocalDate objects are equal
*Deadline: String and LocalDate objects are equal

zhuoyang125 committed:
Merge branch 'branch-C-DetectDuplicates'

zhuoyang125 committed:
Extract Command classes to improve OOP

Extract Command classes to refactor code. Commands such as list,
find, add event, and add deadline inherit from the abstract Command
class. The Command class has an execute() method that will perform
some operation on the task list.

The Parser object includes the createCommand() method to return a
Command that will execute according to the user's input.

zhuoyang125 committed:
Refactor GUI and add delay when exiting

zhuoyang125 committed:
Add user guide

zhuoyang125 committed:
Set theme jekyll-theme-cayman

zhuoyang125 committed:
Create more exceptions to handle additional errors

zhuoyang125 committed:
Merge branch 'master' of https://github.com/zhuoyang125/ip

zhuoyang125 committed:
Improve code style

zhuoyang125 committed:
Remove redundant imports

zhuoyang125 committed:
Add more javadoc comments

zhuoyang125 committed:
Improve code style for test scripts

zhuoyang125 committed:
Remove erroneous assertions

zhuoyang125 committed:
Edit .yml file for continuous integration

zhuoyang125 committed:
Edit parsing command formats

zhuoyang125 committed:
Update user guide

zhuoyang125 committed:
Edit header comments to match coding standard format
